source/controlTypes/state.py

```diff
@@ -28,49 +33,49 @@ def negativeDisplayString(self) -> str:
 			# %s will be replaced with the full identifier of the negated state (e.g. selected).
 			return _("not %s") % self.displayString
 
-	UNAVAILABLE = 0x1
-	FOCUSED = 0x2
-	SELECTED = 0x4
-	BUSY = 0x8
-	PRESSED = 0x10
-	CHECKED = 0x20
-	HALFCHECKED = 0x40
-	READONLY = 0x80
-	EXPANDED = 0x100
-	COLLAPSED = 0x200
-	INVISIBLE = 0x400
-	VISITED = 0x800
-	LINKED = 0x1000
-	HASPOPUP = 0x2000
-	PROTECTED = 0x4000
-	REQUIRED = 0x8000
-	DEFUNCT = 0x10000
-	INVALID_ENTRY = 0x20000
-	MODAL = 0x40000
-	AUTOCOMPLETE = 0x80000
-	MULTILINE = 0x100000
-	ICONIFIED = 0x200000
-	OFFSCREEN = 0x400000
-	SELECTABLE = 0x800000
-	FOCUSABLE = 0x1000000
-	CLICKABLE = 0x2000000
-	EDITABLE = 0x4000000
-	CHECKABLE = 0x8000000
-	DRAGGABLE = 0x10000000
-	DRAGGING = 0x20000000
-	DROPTARGET = 0x40000000
-	SORTED = 0x80000000
-	SORTED_ASCENDING = 0x100000000
-	SORTED_DESCENDING = 0x200000000
-	HASLONGDESC = 0x400000000
-	PINNED = 0x800000000
-	HASFORMULA = 0x1000000000  # Mostly for spreadsheets
-	HASCOMMENT = 0x2000000000
-	OBSCURED = 0x4000000000
-	CROPPED = 0x8000000000
-	OVERFLOWING = 0x10000000000
-	UNLOCKED = 0x20000000000
-	HASNOTE = 0x80000000000
+	UNAVAILABLE = auto()

```

Comment:
Because these are flags that need to work with bitwise operations, I think we need to change the type of this `enum` to something based on `enum.IntFlag` rather than `int`. 
I assume `auto` here is setting values to 1, 2, 3, etc.. instead of 1, 2, 4, 8, etc.

https://docs.python.org/3/library/enum.html#enum.IntFlag

Reply:
It is true that a long time ago there was a requirement for these to be single bit values, as bit-wise operations were performed on them. However, for many years now, NVDA states have been stored and interacted with  as Python sets. Thus bit logic is not at all necessary.
We could ensure that they maintain bit value spacing as Python enums can support this, but this really is no longer necessary.

Reply:
I think it might be worth adding to the changelog entry then:

`controlTypes.State` members are no longer flags, and sets of states should not be determined using bitwise operations.

Reply:
@seanbudd if you are aware of any locations were we still combine these with bitwise operations please let me know. There shouldn't be, but I haven't done an exhaustive search.

Reply:
@feerrenrut - I'm not aware of any

Reply:
I've updated the description with your suggestion @seanbudd